pkg/virt-controller/watch/vmi.go

```diff
@@ -380,6 +380,28 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 			conditionManager.RemoveCondition(vmiCopy, virtv1.VirtualMachineInstanceConditionType(k8sv1.PodReady))
 		}
 
+		// IF the VMI has pod network, Check and patch if IP on VMI is not same as the one on Pod IP
+		for _, network := range vmiCopy.Spec.Networks {
+			if network.NetworkSource.Pod != nil {
+				for i, iface := range vmiCopy.Status.Interfaces {
+					if iface.Name == network.Name {
+						vmiCopy.Status.Interfaces[i].IP = pod.Status.PodIP
+					}
+				}
+				break
+			}
+		}
+		newInterfaces, err := json.Marshal(vmiCopy.Status.Interfaces)
+		if err != nil {
+			return err
+		}
+		log.Log.V(3).Object(vmi).Infof("Patching VMI interfaces")
+		patch := fmt.Sprintf(`{ "op": "replace", "path": "/status/interfaces", "value": %s }`, string(newInterfaces))
+		_, err = c.clientset.VirtualMachineInstance(vmi.Namespace).Patch(vmi.Name, types.JSONPatchType, []byte(fmt.Sprintf("[ %s ]", patch)))
```

Comment:
This would always do a patch, no matter if something changed. You will also have to check if you actually changed something.